### PR TITLE
IPZ:API to update record's ECC

### PR DESCRIPTION
**Pull request by PriyangaRamasamy**
This commit implements an API to update the given record's ECC based on the record's current data.

### CODE REVIEW

File: src/ipz_parser.cpp [hunk 1]
```diff
@@ -631,4 +631,31 @@ types::DbusVariantType IpzVpdParser::readKeywordFromHardware(
     return types::DbusVariantType{
         getKeywordValueFromRecord(l_record, l_keyword, l_recordOffset)};
 }
+
+void IpzVpdParser::updateRecordECC(const auto& i_recordDataOffset,
```
Comment (SunnySrivastava1984) [inline src/ipz_parser.cpp:635]: I could not understand the use case of this API from the given commit message.

**Review by SunnySrivastava1984 — CHANGES_REQUESTED** [review]

File: src/ipz_parser.cpp (hunk 1 above)
Comment (PriyangaRamasamy) [reply src/ipz_parser.cpp:635]: updated the use case in commit message

File: include/ipz_parser.hpp [hunk 2]
```diff
@@ -197,6 +197,24 @@ class IpzVpdParser : public ParserInterface
...
+     * current data.
+     *
+     * @param[in] i_recordDataOffset - Record's data offset
+     * @param[in] i_recordDataLength - Record's data length
+     * @param[in] i_recordECCOffset - Record's ECC offset
+     * @param[in] i_recordECCLength - Record's ECC length
+     *
+     * @throw std::runtime_error on ECC failure.
+     */
+    void updateRecordECC(const auto& i_recordDataOffset,
```
Comment (SunnySrivastava1984) [inline include/ipz_parser.hpp:213]: Please follow camel case in function name and parameters name.
ECC ->Ecc

File: src/ipz_parser.cpp [hunk 3]
```diff
@@ -631,4 +631,31 @@ types::DbusVariantType IpzVpdParser::readKeywordFromHardware(
...
+        const_cast<uint8_t*>(&l_recordECCBegin[0]), &i_recordECCLength);
+
+    if (l_eccStatus != VPD_ECC_OK)
+    {
+        throw std::runtime_error("ECC update failed.");
+    }
+
+    auto l_recordECCEnd = std::next(l_recordECCBegin, i_recordECCLength);
+
+    m_vpdFileStream.exceptions(std::ifstream::badbit | std::ifstream::failbit);
```
Comment (SunnySrivastava1984) [inline src/ipz_parser.cpp:655]: Why it is being added here in this function when it is a class level variable?

File: src/ipz_parser.cpp [hunk 4]
```diff
@@ -631,4 +631,31 @@ types::DbusVariantType IpzVpdParser::readKeywordFromHardware(
     return types::DbusVariantType{
         getKeywordValueFromRecord(l_record, l_keyword, l_recordOffset)};
 }
+
+void IpzVpdParser::updateRecordECC(const auto& i_recordDataOffset,
+                                   const auto& i_recordDataLength,
+                                   const auto& i_recordECCOffset,
+                                   size_t i_recordECCLength)
+{
```
Comment (SunnySrivastava1984) [inline src/ipz_parser.cpp:639]: Whenever these kind of independent function is pushed, at-least make sure that it is being called from somewhere or it is just a prototype.
In this case, it is difficult to figure out if this API is actually giving the desired output or not and you can't add test output of this API to the commit message as well as it is not being called from anywhere.
The approach to break in small commits should be bottom up not the other way round.
Means, the API which needs this should be pushed first with just a comment at the place where this function is supposed to be called or call to an empty prototype of this function. Once that is in, this should be tested and pushed.

**Review by SunnySrivastava1984 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**